cni/pkg/plugin/plugin.go

```diff
@@ -232,8 +232,6 @@ func doAddRun(args *skel.CmdArgs, conf *Config) error {
 		} else {
```

Comment:
I still think we should have the `return` here - just put it after `PushCNIEvent != err` - if we configure the pod for ambient we do not need to perform the subsequent sidecar-only logic or related checks, and can bail early.

Reply:
Thanks for reviewing it. Agreed and updated it.